docs/Language-Tour.md: Differentiate /\ and with (#1060)
dhall-lang/dhall-lang: docs/tutorials/Language-Tour.md

>
> </details>

> > **Exercise:** Louis Reasoner claims that the `with` expression is unnecessary,
> > because he can use the `/\` operator to perform deep record updates. Is he right?

Reviewer: Minor suggestion: perhaps don't use a specific name since not all readers might understand that the name is a stand-in for an arbitrary person (like "John Doe" or "Alice"/"Bob")

```suggestion
> **Exercise:** Your colleague claims that the `with` expression is unnecessary,
> because they can use the `/\` operator to perform deep record updates. Are they right?
```

Author: I am -0 on this suggestion. I think it's a bit of fun if you get the reference (In SICP, Louis Reasoner asked the questions that sounded right but weren't) and a little extra information that you don't _need_ to solve the problem. I think it's also nice to have a bit of connection to the history of our field.

If you still think it's better to use an abstract person, then I'll apply your suggestion and we'll get this shipped.

Reviewer 2: I’m ambivalent about this, but I feel that “Louis Reasoner” might have more value if used multiple times in a consistent way throughout the docs. A single mention like this feels like a bit of a gimmick.

Author: That's fair. If it's not the sort of voice that the project wants in the docs, then it should go.
